File: logdates.m
function full_line = logdates(the_month, day_start, day_end, d)

% function full_line = logdates(the_month, day_start, day_end, d)
% month_start is the month you want
% day_start is the date you want to generate dates from
% day_end is the last date you want to generate dates from
% d = day of the week; M = 1, T = 2, etc, etc
% log dates

% capping the dates generated to end of a month;
if day_end > 31;
    day_end = 31;
    display('You can generate upto the 31st of a month')
end

the_year = 2018;
week = {'M', 'T', 'W', '&Theta;', 'F', 'S', '&Sigma;'};


n=1; % counter for making a new row in text cell

for day = day_start:day_end
    
    day_to_print = sprintf('*%d* ', day);
        day_to_print = sprintf('*%d-%02d-%02d* ', the_year, the_month, day);
    week_to_print = strcat('**',week{d},'**');
    
    d = mod(d+1,8); % only seven days in a week
    if d == 0
        d = 1;
    end
    
    full_line{n,:} = [day_to_print, week_to_print];
    n=n+1;
end

n = n -1;
fileID = fopen('logdates.txt','w');

for m = 1:n
    fprintf(fileID, '%s\n\n', full_line{m,:});
end

fclose(fileID);

end

File: logdates.py
import datetime

def logdates(the_year, the_month):

    # the_year is a number in YYYY format
    # the_month is a number between 1 and 12

    d = int(datetime.date(the_year, the_month, 1).strftime('%w'))
    day_start = 1
    
    if the_month == 2:
        if the_year % 4 == 0:
            day_end = 29
        else:
            day_end = 28
    elif the_month in [1,3,5,7,8,10,12]:
        day_end = 31
    else:
        day_end = 30
    
    week = ['M', 'T', 'W', '&Theta;', 'F', 'S', '&Sigma;']
    full_line = []

    for day in range(day_start,day_end+1):
        day_to_print = '*{:d}-{:02d}-{:02d}*'.format(the_year, the_month, day)
        week_to_print = '**{:s}**'.format(week[d-1])

        d = (d+1) % 7
        if d == 0:
            d = 7

        full_line.append(day_to_print + ' ' + week_to_print)

    for i in full_line:
        print(i)

    fileID = open('log_dates.txt', 'w')

    for line in full_line:
        fileID.write(line)
        fileID.write('\n\n')

    fileID.close()
    
    print('\nText file saved in the same directory as "log_dates.txt"\n')
    
    return 

if __name__ == '__main__':
    the_year = int(input("Year: "))
    the_month = int(input("Month (1-12): "))
    logdates(the_year, the_month)
	


